tasks/rhubarb-audio.js

```diff
@@ -0,0 +1,113 @@
+const fs = require('fs');
+
+let audioDirectory = process.env.npm_package_rhubarbconf_audioDirectory;
+const runrhubarb = process.env.npm_package_rhubarbconf_rhubarbBinary;
+const audioList = process.env.npm_package_rhubarbconf_audioListFile;
+const outputLocation = ensureTrailingSlash(process.env.npm_package_rhubarbconf_outputDirectory) + "lipsync.json";
+let audiofilelist;
+let output = "{";
+
+const { execSync } = require('child_process');
+
+var getallfiles = function(path,fileextension) {
+  const files = [];
+	fs.readdirSync(path,{"withFileTypes":true}).forEach(file => {
+		if(!file.isDirectory() && file.name.indexOf(fileextension) > -1) {
+      var newname = file.name;
+      console.log("found audio file: ",newname);
+      files.push(newname);
+		}
+  })
+  return files;
+}
+
+var getFilesFromList = function(pathtoaudiofile) {
+  // 
+  const files = fs.readFileSync(pathtoaudiofile,'utf8').toString().split("\n");
+  return files;
+}
+
+if(audioDirectory) {
+  if(!audioDirectory.match(/\/$/)){
+    audioDirectory+="/";
+  }
+
+  if(audioList) {
+    audiofilelist = getFilesFromList(audioList);
+  } else {
+    audiofilelist = getallfiles(audioDirectory,'.ogg');
+  }
+  readfiles(audiofilelist);
+
+  output = output.replace(/,\s*$/,"");
+  output +="}";
+  
+  newconfig = JSON.parse(output);
+
+  let fullconfig;
+  try {
+    fullconfig = fs.readFileSync(outputLocation,'utf8');
+    fullconfig = JSON.parse(fullconfig.toString());
+    for(var i in newconfig) {
+     fullconfig[i] = newconfig[i];
+    }
+  } catch {
+    console.log("No existing json file configuration");
+    fullconfig = newconfig;
+  }
+  
+  outputstring = packageJSON(fullconfig,"LIPSYNC"); 
+  if(outputLocation) {
+    fs.writeFileSync(outputLocation, outputstring);
+  }
+  console.log("FINISHED");
+}
+
+function readfiles(audiofilelist) {
+  for(var index in audiofilelist) {
+
+    // strip out the filename in the case that there are captions included. Assume tab delimited.
+    var filename = audiofilelist[index].trim().split("\t")[0];
+
+    console.log('rhubarb ' + audioDirectory + filename);
+    
+    if(runrhubarb) {
+      result = execSync(runrhubarb +' ' + audioDirectory + ensureogg(filename), (err) => {
```

Comment:
rhubarb has an option to feed it a txt file that corresponds with the dialog in an audio file, which gives some better results. I have a process for generating those from a captions.json file that I can add at a later time.

Reply:
The audio text is actually already there, since I am parsing the captions file. As soon as I figure out the syntax to feed it to that function I will add it to the rhubarb call.

Reply:
I've added the functionality to use the processed text from the captions file (which I already had anyways) and feed that in as a file to the rhubarb command. I checked to make sure that it works without error in both cases (with and without the captions available).